packit/base_git.py

```diff
@@ -344,21 +345,52 @@ def add_patches_to_specfile(self, patch_list: List[Tuple[str, str]]) -> None:
             rest_of_the_file = spec_file.read()
             spec_file.seek(last_source_position)
 
-            spec_file.write("\n\n# PATCHES FROM SOURCE GIT:\n")
+            spec_file.write("\n# PATCHES FROM SOURCE GIT:\n")
             for i, (patch, msg) in enumerate(patch_list):
                 commented_msg = "\n# " + "\n# ".join(msg.split("\n")) + "\n"
                 spec_file.write(commented_msg)
-                spec_file.write(f"Patch{i + 1:04d}: {patch}\n")
+                spec_file.write(f"Patch{(i+1):04d}: {patch}\n")
 
-            spec_file.write("\n\n")
+            spec_file.write("\n")
             spec_file.write(rest_of_the_file)
 
-        logger.info(
-            f"Patches ({len(patch_list)}) added to the specfile ({self.absolute_specfile_path})"
-        )
+        logger.info(f"{len(patch_list)} patches added to {self.absolute_specfile_path}")
         self.refresh_specfile()
         self.local_project.git_repo.index.write()
 
+    def specfile_ensure_pnum(self, pnum: int = 1) -> None:
+        """
+        Make sure we use -p1 with %autosetup / %autopatch
```

Comment:
Cannot we use the `Specfile` class for that? (Just an idea, I am not sure about that.)

Reply:
moved and added test

Reply:
Originally, I meant if the rebase-helper parsing cannot help us...;)

But this looks good as well.

Reply:
I know, I was playing with it, but what I saw was:
```python
from rebasehelper.specfile import SpecFile
SpecFile('packit.spec').get_prep_section()
['',
 "cd '/home/jpopelka/rpmbuild/BUILD'",
 "rm -rf 'packitos-0.8.1'",
 "/usr/bin/tar -xof '/packitos-0.8.1.tar.gz'",
 'STATUS=$?',
 'if [ $STATUS -ne 0 ]; then',
 '  exit $STATUS',
 'fi',
 "cd 'packitos-0.8.1'",
 '/usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .',
 '',
 '',
 '# Remove bundled egg-info',
 'rm -rf packitos.egg-info',
 '',
 '']
```
which returns the `%autosetup` macro already expanded, i.e. useless for me.

But now I've just looked at it once more and found
```python
SpecFile('packit.spec').spec_content.section('%prep')                                                                                                                                                                                 
['%autosetup -n %{pypi_name}-%{version}',
 '# Remove bundled egg-info',
 'rm -rf %{pypi_name}.egg-info',
 '']
```
which is what I need, so I'll open another PR.

Reply:
#734